### PR TITLE
CUST-CKFLIGHTF4 Custom Design config

### PR DESCRIPTION
**Pull request by ckflight**
I have created config.h file after completing the CUST-CKFLIGHTF4.config file.
reference: https://github.com/betaflight/unified-targets/pull/1326

However, I do not fully understand the timer mapping and ADC_DMA_OPT line:

B5, B6, B7 and A2 are motor dshot pins which requires timer and dma.
C8 is RGB Led strip pin which requires dshot like protocol so timer and dma is needed as well.
B0 is buzzer pin which requires timer for 4KHz pwm but dma might not be needed (i did not add dma line for this)

```
# timer
timer B05 AF2
# pin B05: TIM3 CH2 (AF2)
timer B06 AF2
# pin B06: TIM4 CH1 (AF2)
timer B07 AF2
# pin B07: TIM4 CH2 (AF2)
timer A02 AF1
# pin A02: TIM2 CH3 (AF1)
timer C08 AF3
# pin C08: TIM8 CH3 (AF3)
timer B00 AF1
# pin B00: TIM1 CH2N (AF1)

# dma
dma ADC 1 1
# ADC 1: DMA2 Stream 4 Channel 0
dma ADC 2 1
# ADC 2: DMA2 Stream 3 Channel 1

dma pin A02 0
# pin A02: DMA1 Stream 1 Channel 3
dma pin B05 0
# pin B05: DMA1 Stream 5 Channel 5
dma pin B06 0
# pin B06: DMA1 Stream 0 Channel 2
dma pin B07 0
# pin B07: DMA1 Stream 3 Channel 2
dma pin C08 0
# pin C08: DMA2 Stream 2 Channel 0
```

As a result how should i set timer mapping lines for config.h file. 4th is the same number of dma pin "pin number" "number". 2nd column is the pin numbe and 1st is for indexing. 3rd column is the part i do not understand. 

```
#define TIMER_PIN_MAPPING \
    TIMER_PIN_MAP( 0, PA2, 1,  0) \
    TIMER_PIN_MAP( 1, PB5, 1,  0) \
    TIMER_PIN_MAP( 2, PB6, 1,  0) \
    TIMER_PIN_MAP( 3, PB7, 1,  0) \
    TIMER_PIN_MAP( 4, PC8, 2,  0)
```

Also Battery and Current adc is available on board how should i set this line:
```
#define ADC1_DMA_OPT        1
```

### CODE REVIEW

**Comment by nerdCopter** [comment]
- `#define ADC1_DMA_OPT        1` is a valid define.
- https://github.com/betaflight/betaflight/blob/master/src/main/pg/timerio.c#L32-L41

recommend inspecting some other config.h files.
someone with deeper understanding will have to reply with more specifics, as i do not know all the technicals.

**Comment by haslinghuis** [comment]
Check https://betaflight.com/docs/development/manufacturer/creating-configuration

Can do it manually or use

image

**Comment by ckflight** [comment]
> Check https://betaflight.com/docs/development/manufacturer/creating-configuration
> 
> Can do it manually or use
> 
> image

I saw that link before but "timer map" command is not active. I can now build betaflight code with github desktop app. I installed arm-none-eabi-gcc and xcode tool. 

Github Desktop App  -> Sublime Text -> Build generates betaflight_4.6.0_STM32F405.hex file.

How can i build using this line:
 
* make TARGET EXTRA_FLAGS="-DUSE_TIMER_MAP_PRINT"

Also how can i build for specific target should i change the lines below in Betaflight/Makefile:

* DEFAULT_TARGET ?= STM32F405
* TARGET    ?= 
* CONFIG    ?=

**Comment by haslinghuis** [comment]
make CKFLIGHF4 EXTRA_FLAGS="-DUSE_TIMER_MAP_PRINT"

**Review by ckflight — COMMENTED** [review]
It is working. I have build for CKFLIGHTF4 and tested on drone. Everything works like a charm :)

**Review by ckflight — COMMENTED** [review]
It fully checked and working

File: configs/CKFLIGHTF4/config.h [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+#define BARO_SPI_INSTANCE SPI1
+
+#define BEEPER_INVERTED
+
+#define USE_SDCARD_SPI
+#define SDCARD_SPI_INSTANCE SPI3
+#define DEFAULT_BLACKBOX_DEVICE     BLACKBOX_DEVICE_SDCARD
+
+#define SYSTEM_HSE_MHZ 8
+#define DASHBOARD_I2C_INSTANCE (I2CDEV_2)
```
Comment (haslinghuis) [inline configs/CKFLIGHTF4/config.h:103]: Won't be needed
```suggestion
```

File: configs/CKFLIGHTF4/config.h [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    TIMER_PIN_MAP(3, PA2, 1, 0) \
+    TIMER_PIN_MAP(4, PC8, 2, 0) \
+    TIMER_PIN_MAP(5, PB0, 1, -1)
+
+
+#define ADC1_DMA_OPT        1
+
+#define MAG_I2C_INSTANCE (I2CDEV_2)
+
+#define USE_BARO
```
Comment (haslinghuis) [inline configs/CKFLIGHTF4/config.h:93]: Moving on top
```suggestion
```

File: configs/CKFLIGHTF4/config.h [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+#define FC_TARGET_MCU     STM32F405
+
+#define BOARD_NAME        CKFLIGHTF4
+#define MANUFACTURER_ID   CUST
+
+#define USE_GYRO
+#define USE_GYRO_SPI_ICM20602
+#define USE_ACC
+#define USE_ACC_SPI_ICM20602
+#define USE_BARO_SPI_MS5611
```
Comment (haslinghuis) [inline configs/CKFLIGHTF4/config.h:33]:
```suggestion
#define USE_BARO
#define USE_BARO_SPI_MS5611
```

**Review by haslinghuis — APPROVED** [review]

File: configs/CKFLIGHTF4/config.h [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+#define MAG_I2C_INSTANCE (I2CDEV_2)
+
+#define BARO_SPI_INSTANCE SPI1
+
+#define BEEPER_INVERTED
+
+#define USE_SDCARD_SPI
+#define SDCARD_SPI_INSTANCE SPI3
+#define DEFAULT_BLACKBOX_DEVICE     BLACKBOX_DEVICE_SDCARD
+
```
Comment (haslinghuis) [inline configs/CKFLIGHTF4/config.h:101]:
```suggestion
```

File: configs/CKFLIGHTF4/config.h [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+#define BEEPER_INVERTED
+
+#define SYSTEM_HSE_MHZ 8
+
+#define GYRO_1_SPI_INSTANCE SPI1
+#define GYRO_1_ALIGN CW0_DEG
+#define GYRO_1_ALIGN_YAW 0
+
+#define MAX7456_SPI_INSTANCE SPI2
+
```
Comment (haslinghuis) [inline configs/CKFLIGHTF4/config.h:105]:
```suggestion

#define USE_SDCARD_SPI
#define SDCARD_SPI_INSTANCE SPI3
#define DEFAULT_BLACKBOX_DEVICE     BLACKBOX_DEVICE_SDCARD
```

**Review by haslinghuis — APPROVED** [review]

**Comment by ckflight** [comment]
Thank you so much for your help, contributions and corrections.

**Review by KarateBrot — APPROVED** [review]

**Pull request merged**